src/lib/telemetry/TelemetryReporter.ts

```diff
@@ -31,16 +38,58 @@ export class TelemetryReporter {
   }
 }
 
+const lockfilePromisified = (lockName, options) => {
+  return new Promise((resolve, reject) => {
+    lockfile.lock(lockName, options, err => {
+      err ? reject(err) : resolve()
+    })
+  })
+}
+
 const start = async () => {
   try {
     const store = new TelemetryLocalStore(process.argv[2])
-    const telemetryObj = JSON.parse(process.argv[3])
+
+    // Send general toolbelt errors
+    const telemetryObjFilePath = process.argv[3]
+    const telemetryObj = await readJson(telemetryObjFilePath)
     const reporter = TelemetryReporter.getTelemetryReporter()
     await reporter.reportErrors(telemetryObj.errors)
+    await remove(telemetryObjFilePath)
+
+    // Send TelemetryReport errors
+    const lockfileName = 'reporter.lock'
+    const lockfilePath = join(TelemetryReporter.ERRORS_DIR, lockfileName)
+    await ensureDir(TelemetryReporter.ERRORS_DIR)
+    await lockfilePromisified(lockfilePath, {})
+    const metaErrorsFiles = (await readdir(TelemetryReporter.ERRORS_DIR)).filter(fileName => {
+      return fileName !== lockfileName
+    })
+    await Promise.all(
+      metaErrorsFiles.map(async metaErrorsFile => {
+        const metaErrorsObject = await readJson(join(TelemetryReporter.ERRORS_DIR, metaErrorsFile))
+        await reporter.reportErrors(metaErrorsObject)
+        await remove(metaErrorsFile)
+      })
+    )
+    lockfile.unlock(lockfilePath)
+
     store.setLastRemoteFlush(Date.now())
     process.exit()
   } catch (err) {
-    console.log(err)
+    const errorFilePath = `${TelemetryReporter.ERRORS_DIR}/${randomBytes(8).toString('hex')}.json`
```

Comment:
Maybe we should call `unlock` here as well? If we use the `FileLock` class it could store a boolean `isLocked` indicating that the lock was called successfully, then `unlock` would call `lockfile.unlock` only in case `isLocked` is true to avoid errors

Reply:
Actually I think we should call `unlock` at the end of this catch